snet_cli/mpe_channel_command.py

```diff
@@ -32,29 +32,67 @@ def get_service_spec_dir(self, org_id, service_id):
+        tmp = tempfile.NamedTemporaryFile(delete=False)
+        pickle.dump( channels_dict, open( tmp.name, "wb" ) )
+        os.rename(tmp.name, self._get_channels_info_file(org_id, service_id))
+
+    def _get_initialized_channels_dict_for_service(self, org_id, service_id):
+        '''return {channel_id: channel}'''
+        fn = self._get_channels_info_file(org_id, service_id)
+        if (os.path.isfile(fn)):
+            return pickle.load( open( fn, "rb" ) )
+        else:
+            return {}
+
+    def _get_initialized_channels_for_service(self, org_id, service_id):
+        '''return [channel]'''
+        channels_dict = self._get_initialized_channels_dict_for_service(org_id, service_id)
+        return list(channels_dict.values())
 
-    def _save_channel_info(self, channel_dir, channel_info):
-        fn = os.path.join(channel_dir, "channel_info.pickle")
-        pickle.dump( channel_info, open( fn, "wb" ) )
+    def _get_service_info_file(self, org_id, service_id):
+        return os.path.join(self._get_service_base_dir(org_id, service_id), "service_info.pickle")
 
-    def _read_channel_info(self, org_id, service_id, channel_id):
-        fn = os.path.join(self.get_channel_dir(org_id, service_id, channel_id), "channel_info.pickle")
-        channel_info = pickle.load( open( fn, "rb" ) )
-        channel_info["channelId"] = channel_id
-        return channel_info
+    def _save_service_info(self, org_id, service_id, service_info):
+        fn = self._get_service_info_file(org_id, service_id)
+        pickle.dump( service_info, open( fn, "wb" ) )
+
+    def _read_service_info(self, org_id, service_id):
+        fn = self._get_service_info_file(org_id, service_id)
+        return pickle.load( open( fn, "rb" ) )
+
+    def is_service_initialized(self):
+        return os.path.isfile(self._get_service_info_file(self.args.org_id, self.args.service_id))
 
     def _check_mpe_address_metadata(self, metadata):
         """ we make sure that MultiPartyEscrow address from metadata is correct """
         mpe_address = self.get_mpe_address()
         if (str(mpe_address).lower() != str(metadata["mpe_address"]).lower()):
             raise Exception("MultiPartyEscrow contract address from metadata %s do not correspond to current MultiPartyEscrow address %s"%(metadata["mpe_address"], mpe_address))
 
-    def _init_new_service_from_metadata(self, service_dir, metadata):
+    def _init_or_update_service_if_needed(self, metadata, service_registration):
+        # if service was already initialized and metadataURI hasn't changed we do nothing
+        if self.is_service_initialized():
+            if self.is_metadataURI_has_changed(service_registration):
+                self._printerr("# Service with org_id=%s and service_id=%s was updated"%(self.args.org_id, self.args.service_id))
+                self._printerr("# ATTENTION!!! price or other paramaters might have been changed!\n")
```

Comment:
But why here? This message is printed only in case then service was reinitialized in case then change in metadata was detected... And normally all payment channels should still be valid unless service provider change groupId or payment_address which we actively prevent. If you try to change groupId or payment_address with ```snet service update-metadata``` you will get an error..
Of course service provider still can change group_id and make all channels invalid but it should be considered as malicious behavior..

Reply:
I thought `group_id` was generated using `endpoint` as part of its base64 bytes construction.
Just to be sure, can service provider update `endpoint` without update `group_id`? If yes, so me comment above can be ignored.

Reply:
```group_id``` is totally independent from endpoint. Service provider can update endpoint without updating ```group_id```, but we don't have functions for it yet :( (we have only function for adding endpoint to metadata but not for removing it). Probably we should add such a function...